hdfs3/core.py

```diff
@@ -333,7 +337,8 @@ def rm(self, path, recursive=True):
         if not self.exists(path):
             raise FileNotFoundError(path)
         out = _lib.hdfsDelete(self._handle, ensure_byte(path), bool(recursive))
-        return out == 0
+        if out != 0:
+            raise IOError('Remove failed')
```

Comment:
Can we add the path to these error messages?  Sometimes the user gets these messages when the call was made by some other library, deep in the users code.  They may not know what the errant file was.